src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java

```diff
@@ -110,8 +113,12 @@ public GetProgramResponse getProgram(GetProgramRequest request) {
   public UpdateProgramResponse updateProgram(UpdateProgramRequest request) {
     val program = request.getProgram();
     val updatingProgram = programConverter.programToProgramEntity(program);
+    val programToUpdate = programService.getProgram(updatingProgram.getShortName(), false);
+
+    updateMembershipPermission(programToUpdate, updatingProgram);
```

Comment:
should this just live inside `updateProgram` method of `ProgramService`? so we don't have to change `updateProgram`'s signature?